### ISSUE TITLE
Bug: no-dupe-keys ignore dynamic keys duplicates

### ISSUE THREAD

**Issue by kaaboaye** [eslint/eslint#16418]
### Environment

ESLint version: v8.25.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
/*eslint no-dupe-keys: "error"*/
```
</details>

```js
// true negative
const a = {
  a: 1,
  b: 2
}

// true positive
const b = {
  a: 1,
  a: 1
}

const Enum = {
  Some: "SOME",
  None: "NONE"
}

// false negative
const c = {
  [Enum.Some]: 1,
  [Enum.Some]: 2,
  [Enum.None]: 3
}
```

### What did you expect to happen?

To detect dynamic keys duplicates

### What actually happened?

It didn’t 

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @kaaboaye, thanks for the issue!

Detecting duplicate dynamic keys is not as reliable as detecting duplicate static keys, due to side effects that can occur during the evaluation of object initializers.

`no-dupe-keys` currently ignores computed property names by design, so this is not a bug.

I'm closing this issue since it's submitted as a bug report. If you have an idea about how detecting duplicate dynamic keys could work, please submit the proposal as a [rule change request](https://github.com/eslint/eslint/issues/new?assignees=&labels=enhancement%2Crule&template=rule-change.yml&title=Rule+Change%3A+%28fill+in%29).
